torch/quantization/ns/graph_matcher.py

```diff
@@ -87,12 +88,47 @@ def get_non_matchable_modules() -> Set[Callable]:
         torch.quantization.FakeQuantizeBase,
     ])
 
-class _NSGraphMatchableNodesIterator:
+def get_reversed_fusions() -> Set[Tuple[Callable, Callable]]:
+    """
+    Set of potential fusions, in reverse order.  The order is reversed
+    to match how fusion patterns are defined in quantization code.
+    """
+    return set([
+        (F.relu, F.linear),
+    ])
+
+# TODO(future PR): we should see if we can reuse quantization's fusion
+# patterns here.
+def end_node_matches_reversed_fusion(
+    end_node: Node,
+    reversed_fusion: Tuple[Callable, Callable],
+) -> bool:
+    """
+    Returns true if a pattern ending with `end_node` matches
+    the fusion pattern.
+    """
+    if end_node.op == 'call_function':
+        cur_node = end_node
+        for fusion_idx in range(len(reversed_fusion)):
+            cur_fusion_op = reversed_fusion[fusion_idx]
+            if cur_node.target != cur_fusion_op:
+                return False
+            if len(cur_node.args) > 0 and isinstance(cur_node.args[0], Node):
+                cur_node = cur_node.args[0]
+            else:
+                return False
+        return True
+    # TODO(future PR): handle call_module
+    return False
+
+
+class _NSGraphMatchableSubgraphsIterator:
     """
     Iterates through the graph of gm, starting with the output nodes
     and continuing backwards.
-    1. Returns matchable nodes, in order
-    2. Skips over non-matchable nodes
+    1. Returns matchable subgraphs, in order. A subgraph is defined by
```

Comment:
Sorry, dont follow this logic: Where is the sub-graph specified in the init? I dont see the end_node being defined.

Reply:
this is in the `__next__` function.  We pop a potential end node off the stack.  If the end node is matchable, we try to peek backwards to check for a potential fusion pattern.  We return (end_node, end_node) if there is no fusion pattern, or (start_node, end_node) if there is a fusion pattern ending at end_node.